转录组入门(2)：读文章拿到测序数据
本系列课程学习的文章是：AKAP95 regulates splicing through scaffolding RNAs and RNA processing factors. Nat Commun 2016 Nov 8;7:13347. PMID: 27824034 很容易在文章里面找到数据地址GSE81916 这样就可以下载sra文件

作业，看文章里的methods部分，把它用到的软件和参数摘抄下来，然后理解GEO/SRA数据库的数据存放形式，把规律和笔记发在论坛上面

Bioinformatic analyses
---------

We obtained 35–55 million of paired 51 bp reads for each sample. The investigator was blinded to sample identities when performing bioinformatic analyses. All the reads were mapped to the human reference genome (GRCh37/hg19) using TopHat (v2.0.13) with a gene transfer file (GTF version GRCh37.70). Low quality mapped reads (MQ>30) were removed from the analysis. The mean insert sizes and the s.d.’s were calculated using Picard-tools (v1.126) (http://broadinstitute.github.io/picard/).
For mRNA-Seq, read count tables were generated using HTSeq (v0.6.0)53 and deferential expression (DE) analysis of genes were performed using DESeq (v3.0)54.
Deferential exon usage was performed using DEXSeq (v3.1)29.
Two independent biological repeats for control and KD were included in both DEseq and DEXseq.

For 293 cells, the mRNA-seq results of the control samples include
(1) those from the doxycycline-treated parental Flp-In T-REx 293 cells by us and
(2) those from the doxycycline-treated control Flp-In T-REx 293 cells performed by another group unrelated to us (sample GSM1095127 in GSE44976)31, and the two AKAP95 KD samples were from the two different clones (miR#8 and miR#12) of Flp-In T-REx 293 cells stably expressing AKAP95 microRNAs after doxycycline treatment. For mouse ES cells, the two biological repeats for both control and shRNA#1-mediated Akap95 KD were independently generated by two different persons in our lab.

The read per million normalized BigWig files were generated using BEDTools (v2.17.0)55 and bedGraphToBigWig tool (v4). All the downstream statistical analyses and generating plots were performed in R (v3.1.1) (http://www.r-project.org/). Gene ontology analysis was performed at DAVID (http://david.ncifcrf.gov/)

For RIP-Seq, mapped files were passed to Model-based Analysis of ChIP-Seq (MACS) (v1.4.2 20120305)56 for peak calling and HOMER (vv3.12, 6-8-2012) (http://homer.salk.edu/homer/)57 was used for Motif finding. Bedgraph files generated by MACS were then normalized based on read per million and converted to BigWig files using bedGraphToBigWig. The input RNA-seq files were not subject to MACS. The genomic regions for annotation (intergenic regions, exons, UTRs, distal and proximal introns) were extracted from the GTF file used above. Peak annotations and coverage calculations for these genomic regions were performed using Bedtools and SAMtools (v0.1.18)58, respectively. Profile plots were generated by ngs.plot (v2.47)59.

